libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java

```diff
@@ -65,8 +68,20 @@ public SoqlSyncDownTarget(String query) {
         super();
         this.queryType = QueryType.soql;
         this.query = query;
+        addIdFieldIfRequired();
 	}
-	
+
+    private void addIdFieldIfRequired() {
+        if (!TextUtils.isEmpty(query)) {
+            final String idFieldName = getIdFieldName();
+
+            // Inserts the mandatory 'Id' field if it doesn't exist.
+            if (!query.contains(idFieldName)) {
+                query = query.replaceFirst("([sS][eE][lL][eE][cC][tT] )", "select " + idFieldName + ", ");
```

Comment:
I guess it should work even with subqueries (https://developer.salesforce.com/docs/atlas.en-us.200.0.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select.htm) thanks to the replaceFirst.